### ISSUE TITLE
FastNoiseLite for NoiseTexture3D always returns zero

### ISSUE THREAD

**Issue by Makaque** [godotengine/godot#95382]
### Tested versions

v4.2.2.stable.official [15073afe3]

### System information

Godot v4.2.2.stable - Windows 10.0.19045 - GLES3 (Compatibility) - ANGLE (AMD, AMD Radeon (TM) R9 M360 (0x0000682B) Direct3D11 vs_5_0 ps_5_0, D3D11-27.20.1034.6) () - Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz (8 Threads)

### Issue description

 I set up a mesh with a ShaderMaterial. The shader takes a sampler3D, to which I'm feeding a Simplex Smooth noise texture generated with the built-in FastNoiseLite.

For a simple test I have this

```
vec4 tex = texture(landmass_texture, VERTEX);
VERTEX += (NORMAL * tex.xyz);
```

But it's always zero. No error is ever thrown, but it's always zero. 2D textures work, but 3D doesn't.

### Steps to reproduce

- Start project
- add CSGMesh3D root node
- In the inspector, add a new ShereMesh and a new ShaderMaterial
- In the Material properties, add a new Spatial Shader
- Write the shader 

```
shader_type spatial;
uniform sampler3D noise_texture;

void vertex() {
	vec4 tex = texture(noise_texture, VERTEX);
	VERTEX += (NORMAL * tex.xyz);
}
```

- In the inspector, in Shader Parameters create a new NoiseTexture3D
- In the NoiseTexture3D properties, for noise, create a new FastNoiseLite 
- There should be visible changes to the mesh, but `NORMAL * tex.xyz` is always zero.

### Minimal reproduction project (MRP)

[Noise3D.zip](https://github.com/user-attachments/files/16571953/Noise3D.zip)

**Comment by AThousandShips** [comment]
Please explain what the issue is, what is expected and what are your results?

For me (in 4.3 [88f3b5f9d52f740b24fabfb8bc01b8b7026ba279]) it renders with a noisy lumpy shape, with a 3D noise texture

**Comment by Grublady** [comment]
Also unable to reproduce on 4.3-rc3; it renders in a noisy 3D shape.

**Comment by Makaque** [comment]
I see the default sphere, unchanged, with a 1 unit diameter. 
This problem seems to only appear in version 4.2.2 and only when Compatibility is chosen for the renderer. Fixed in 4.3-rc3.

**Comment by AThousandShips** [comment]
Would need to identify what fixed it in 4.3 to consider for cherry picking for a future 4.2.x, otherwise this can be closed as it's fixed
